### PR TITLE
Ports to other browser vendors

### PR DESCRIPTION
**Pull request by gampleman**
@smcgivern can you take a look?

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
+  chrome.tabs.onUpdated.addListener(function(tabId, changeInfo) {
+    if (changeInfo.status !== 'complete') { return; }
 
-  chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected;'}, function(res) {
-    if (res && !res[0]) {
-      chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected = true;'});
+    if (PLATFORM == 'chrome') {
+      chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected;'}, function(res) {
+        if (res && !res[0]) {
+          // chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected = true;'});
```
Comment (smcgivern) [inline src/background.js]: Delete or uncomment.

File: src/background.js [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+          chrome.tabs.executeScript(tabId, {file: 'src/ui-previewer-button.js'});
+        }
+      });
+    } else if(PLATFORM == 'firefox') {
```
Comment (smcgivern) [inline src/background.js:11]: The platform is only Chrome or Firefox in this context, might be worth keeping this to be explicit anyway though.

File: README.md [hunk 3]
```diff
@@ -28,6 +28,54 @@ steps:
    about the format, visit the extension's options page, where you can also configure
    repos manually.
 
+## Building
+
+Make sure you have npm in your path. Run `make`. This will build as much as can be
+reasonably automated for each supported platform. Unfortunately, to finish building,
+test and deploy, you will need to do a few manual steps:
+
+#### Firefox
```
Comment (smcgivern) [inline README.md:37]: Are these two header levels down from the previous one?

**Comment by smcgivern** [comment]
Looks good, love a good makefile. Looks like the Travis build might need updating to point to the new Chrome unpacked extension directory.

**Comment by gampleman** [comment]
Yeah I love how fast make is.

BTW also added a nice little installation and configuration website: http://rightscale.github.io/ui-previewer-button/

**Comment by smcgivern** [comment]
:heart_eyes: 

That's awesome.

**Pull request merged**